src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -39,169 +37,178 @@ public static async Task Main(string[] args)
-                foreach (var workspaceDiagnostic in workspace.Diagnostics)
+            if (incrementalAnalyzerRunner.HasAnalyzers)
+            {
+                if (!string.IsNullOrEmpty(options.ProfileRoot))
                 {
-                    if (workspaceDiagnostic.Kind == WorkspaceDiagnosticKind.Failure)
-                    {
-                        Console.WriteLine(workspaceDiagnostic.Message);
-                    }
+                    ProfileOptimization.StartProfile(nameof(Microsoft.CodeAnalysis.SolutionCrawler.IIncrementalAnalyzer));
                 }
 
-                foreach (var projectId in projectIds)
+                await incrementalAnalyzerRunner.RunAsync(cancellationToken).ConfigureAwait(false);
+            }
+
+            if (diagnosticAnalyzerRunner.HasAnalyzers)
+            {
+                if (!string.IsNullOrEmpty(options.ProfileRoot))
                 {
-                    solution = solution.WithProjectAnalyzerReferences(projectId, ImmutableArray<AnalyzerReference>.Empty);
+                    ProfileOptimization.StartProfile(nameof(DiagnosticAnalyzerRunner));
                 }
 
-                Console.WriteLine($"Loaded solution in {stopwatch.GetSummary(preciseMemory: true)}");
+                await diagnosticAnalyzerRunner.RunAllAsync(cancellationToken).ConfigureAwait(false);
+            }
 
-                if (options.ShowStats)
+            if (codeRefactoringRunner.HasRefactorings)
+            {
+                if (!string.IsNullOrEmpty(options.ProfileRoot))
                 {
-                    stopwatch = PerformanceTracker.StartNew();
-
-                    List<Project> projects = solution.Projects.Where(project => project.Language == LanguageNames.CSharp || project.Language == LanguageNames.VisualBasic).ToList();
+                    ProfileOptimization.StartProfile(nameof(CodeRefactoringRunner));
+                }
 
-                    Console.WriteLine("Number of projects:\t\t" + projects.Count);
-                    Console.WriteLine("Number of documents:\t\t" + projects.Sum(x => x.DocumentIds.Count));
+                await codeRefactoringRunner.RunAsync(cancellationToken).ConfigureAwait(false);
+            }
+        }
 
-                    var statistics = GetSolutionStatistics(projects, cancellationToken);
+        private static async Task ShowCompilerDiagnosticsAsync(Solution solution, CancellationToken cancellationToken)
+        {
+            var projectIds = solution.ProjectIds;
 
-                    Console.WriteLine("Number of syntax nodes:\t\t" + statistics.NumberofNodes);
-                    Console.WriteLine("Number of syntax tokens:\t" + statistics.NumberOfTokens);
-                    Console.WriteLine("Number of syntax trivia:\t" + statistics.NumberOfTrivia);
+            foreach (var projectId in projectIds)
+            {
+                solution = solution.WithProjectAnalyzerReferences(projectId, ImmutableArray<AnalyzerReference>.Empty);
+            }
```

Comment:
📝 This was originally unconditional

Reply:
The new solution was never applied back to the workspace, so it had no effect on any of the runners.

Reply:
Also, DiagnosticAnalyzerRunner does have this 
```cs
CompilationWithAnalyzers compilationWithAnalyzers = newCompilation.WithAnalyzers(analyzers, new CompilationWithAnalyzersOptions(workspaceAnalyzerOptions, null, analyzerOptionsInternal.RunConcurrent, logAnalyzerExecutionTime: true, reportSuppressedDiagnostics: analyzerOptionsInternal.ReportSuppressedDiagnostics));

```